### PR TITLE
Adding retrofit

### PR DESCRIPTION
**Pull request by raymondpoling**
Adds retrofit, and a skeleton concept of the repository (just a layer that merges data retrieval and caching; no effort to make it correct to pattern will be attempted).

### CODE REVIEW

File: app/src/main/java/org/americanairlines/our1grouprestaurantmapproject/MapsActivity.kt [hunk 1]
```diff
@@ -20,7 +20,7 @@ class MapsActivity : AppCompatActivity(), OnMapReadyCallback {
         super.onCreate(savedInstanceState)
```
Comment (James-H2o) [inline app/src/main/java/org/americanairlines/our1grouprestaurantmapproject/MapsActivity.kt]: This file can be deleted

File: app/src/main/java/org/americanairlines/our1grouprestaurantmapproject/model/NearbyPlacesModel.kt [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+/*
+ * This is a place to put data our app will actually use (as well as providing a simple caching object.)
+ * this class should be modified to reflect the data we want.
+ * The primary key is nullable since, well, we cannot guarentee
+ */
+
+@Entity
+data class NearbyPlacesModel(
+        @PrimaryKey(autoGenerate = true) val id : Int?,
```
Comment (James-H2o) [inline app/src/main/java/org/americanairlines/our1grouprestaurantmapproject/model/NearbyPlacesModel.kt:15]: Primary key should not be nullable

File: .idea/.gitignore [hunk 3]
```diff
@@ -1,3 +0,0 @@
-# Default ignored files
```
Comment (James-H2o) [inline .idea/.gitignore:1]: Why was this file deleted?

**Comment by James-H2o** [comment]
Please review my comments and consider making changes before merging

File: app/src/main/java/org/americanairlines/our1grouprestaurantmapproject/MapsActivity.kt (hunk 1 above)
Comment (raymondpoling) [reply app/src/main/java/org/americanairlines/our1grouprestaurantmapproject/MapsActivity.kt]: Agreed

File: app/src/main/java/org/americanairlines/our1grouprestaurantmapproject/model/NearbyPlacesModel.kt (hunk 2 above)
Comment (raymondpoling) [reply app/src/main/java/org/americanairlines/our1grouprestaurantmapproject/model/NearbyPlacesModel.kt:15]: If room complains, then I'll change it, but only room should assign ids.

File: .idea/.gitignore (hunk 3 above)
Comment (raymondpoling) [reply .idea/.gitignore:1]: I usually do no keep .idea/* files because across build environments they can cause very nasty, hard to find bugs due to minor environment changes.

**Comment by raymondpoling** [comment]
Review changes have been pushed.
